### DISCUSSION TITLE
PockeHost 0.8.0 Released 🎉

### DISCUSSION THREAD

**Discussion by benallfree** [pockethost/pockethost#229] [Show and tell]
# PocketHost 0.8.0 Release Notes

[PocketHost](https://pockethost.io) is the zero-config 'PocketBase in the cloud' platform. Thank you for everyone's continued support for this project.

## What's New

There are several MAJOR changes in this release.

### New Instance UI

The instance UI in the dashboard has been redesigned to improve organization and clarity.

<img width="930" alt="image" src="https://github.com/benallfree/pockethost/assets/1068356/66d5612e-d207-4e20-b57f-cc0b7b010e66">

### Danger Zone

A new accordion tab, _Danger Zone_, has been added. This tab contains potentially harmful or destructive settings for your PocketBase instance.

<img width="403" alt="image" src="https://github.com/benallfree/pockethost/assets/1068356/e8a20b4b-ca8c-4cac-bc3b-3bcfac53577c">

### Rename Instance

Instances can now be renamed.

<img width="415" alt="image" src="https://github.com/benallfree/pockethost/assets/1068356/9e844056-2645-433b-b20d-bac41e25ec5b">

### Instance Permalink

Every instance is accessible from `https://<uuid>.pockethost.io`

### Maintenance Mode

You can now put your PocketHost instance in Maintenance Mode. Maintenance mode will shut down any running copies of the instance and keep it shut down until you take it out of maintenance mode. This is helpful for backups and restores in particular.

<img width="317" alt="image" src="https://github.com/benallfree/pockethost/assets/1068356/6f5c93bc-15db-45e1-b711-5cd570795253">

### Version range locking

PocketHost has supported semantic version ranges for a long time. Now, you can edit the version of PocketBase for your instance.

<img width="472" alt="image" src="https://github.com/benallfree/pockethost/assets/1068356/e28c7c54-44dd-4f1a-b69c-1495c151bd3b">

### Stability Improvements

In this release, the code managing the launching and shutting down of instances has been completely reworked. The result is dramatically improved stability and performance.

### Stress Testing

Each release of PocketHost is now tested with 500 simultaneous requests to hundreds of different PocketBase instances. Stress testing revealed some rare error conditions which have been corrected.

## Change log

- fix: instance store and realtime responsiveness
- fix: instance shutdown while in unhealthy state
- fix: frontend logger refactor
- enh: dashbaord for large number of instances
- Feat: mega stability overhaul, maintenance mode, rename, respond by uuid
- refactor: version change
- enh: process killing enhancements
- enh: version download enhancements
- refactor: rpc commands
- enh: autocancel behavior for instance client mixin
- enh: downloadManager serialization guard
- enh: SqliteService async serialization enhancements
- enh: start central pocketbase at daemon port base
- fix: createInvocation autocancel key
- enh: logging output
- fix: remove async promise handler pattern from InstanceLoggerService
- chore: FtpService logging and code formatting enhancements
- fix: handle preflight exceptions in tryCatch
- enh: SqliteLogger logging
- chore: remove backups table from db
- chore: remove RPC commands support from db
- fix: remove async promise handler (won't catch errors)
- fix: undefined value in TimerManager
- enh: cleanupManager async cleanups and singleton shutdown
- enh: safeCatch logging and handling
- enh: terminal colors for logging
- enh: RPC hygen templates
- fix: faq link
- enh: instance overview formatting
- enh: running->version
- enh: organize instance info in accordion
- fix: secrets form dark mode
- enh: add admin link to instance title
- enh: bootstrap 5.3 update, hljs dark mode
- fix: remove WorkerLogFields reference
- cos: tiny button refactor & reduce size of code copy button
- chore: get rid of worker log fields
- big: refactor instancemanager for proper shutdown when pocketbase fails
- bug: reset instances on boot
- bug: request hangs if pockethost doesn't launch
- docs

## Stats and Info

- 331 github stars
- ~2000 registered users
- ~2700 instances created
- ~1.5 million instance invocations to date (# times PocketBase executable has been launched)
- ~400 instances active and used in the past week

Check out the [FAQ](https://pockethost.gitbook.io/manual/overview/faq) and [roadmap](https://pockethost.gitbook.io/manual/overview/roadmap) for more details.

**Comment by RA9** [comment]
Great job, @benallfree! I am grateful for Pockethost.

**Comment by renatocaliari** [comment]
Great news! 🚀 Congratulations on the work and thank you for this platform @benallfree
